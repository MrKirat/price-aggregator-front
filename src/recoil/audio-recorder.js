import { atom } from 'recoil';

const isAudioRecorderEnabledState = atom({
    key: 'isAudioRecorderEnabledState',
    default: false
});

export default isAudioRecorderEnabledState;