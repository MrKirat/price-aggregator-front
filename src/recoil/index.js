export { default as isAudioRecorderEnabledState } from './audio-recorder';
export { default as isCameraEnabledState } from './camera';
export { default as searchStringState } from './search';
export { default as productListState } from './products';
export { default as isLoadingState } from './loading';