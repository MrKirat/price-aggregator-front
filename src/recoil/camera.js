import { atom } from 'recoil';

const isCameraEnabledState = atom({
    key: 'isCameraEnabledState',
    default: false
});

export default isCameraEnabledState;