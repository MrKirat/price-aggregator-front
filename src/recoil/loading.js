import { atom } from 'recoil';

const isLoadingState = atom({
    key: 'isLoadingState',
    default: false
});

export default isLoadingState;