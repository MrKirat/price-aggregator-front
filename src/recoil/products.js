import { atom } from 'recoil';

const productListState = atom({
    key: 'productListState',
    default: []
});

export default productListState;