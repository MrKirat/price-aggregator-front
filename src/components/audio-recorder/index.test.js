import React from 'react';

import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AudioRecorder from './index';
import { RecoilRoot } from 'recoil';

it('renders without crashing', () => {
  const output = mount(<RecoilRoot><AudioRecorder /></RecoilRoot>);
  expect(shallowToJson(output)).toMatchSnapshot();
  expect(output.find('svg')).toHaveLength(1);
});

it('renders with icon', () => {
  const output = mount(<RecoilRoot><AudioRecorder /></RecoilRoot>);
  expect(output.find('svg')).toHaveLength(1);
});

it('renders with AudioRecorder', () => {
  const output = mount(<RecoilRoot><AudioRecorder /></RecoilRoot>);
  expect(output.find('AudioRecorder')).toHaveLength(1);
});