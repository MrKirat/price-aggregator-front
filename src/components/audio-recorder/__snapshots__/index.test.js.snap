// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RecoilRoot>
  <Batcher
    setNotifyBatcherOfChange={[Function]}
  />
  <AudioRecorder>
    <ForwardRef
      className="makeStyles-icon-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(SvgIcon))
        className="makeStyles-icon-1"
        onClick={[Function]}
      >
        <ForwardRef(SvgIcon)
          className="makeStyles-icon-1"
          classes={
            Object {
              "colorAction": "MuiSvgIcon-colorAction",
              "colorDisabled": "MuiSvgIcon-colorDisabled",
              "colorError": "MuiSvgIcon-colorError",
              "colorPrimary": "MuiSvgIcon-colorPrimary",
              "colorSecondary": "MuiSvgIcon-colorSecondary",
              "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
              "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
              "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
              "root": "MuiSvgIcon-root",
            }
          }
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-1"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
            />
          </svg>
        </ForwardRef(SvgIcon)>
      </WithStyles(ForwardRef(SvgIcon))>
    </ForwardRef>
  </AudioRecorder>
</RecoilRoot>
`;
